app/screens/home/channel_list/categories_list/categories/unreads/index.ts

```diff
@@ -50,42 +55,62 @@ const mostRecentFirst = (a: MyChannelModel, b: MyChannelModel) => {
  * Unreads, Mentions, and Muted Mentions Only
  *
  * Mentions on top, then unreads, then muted channels with mentions.
- * Secondary sorting within each of those is by recent posting.
+ * Secondary sorting within each of those is by recent posting or by recent root post if CRT is enabled.
  */
 
-const filterAndSortMyChannels = ([myChannels, notifyProps]: [MyChannelModel[], NotifyProps]): MyChannelModel[] => {
+type FilterAndSortMyChannelsArgs = [
+    MyChannelModel[],
+    NotifyProps,
+    Record<string, number> | undefined
+]
+
+const filterAndSortMyChannels = ([myChannels, notifyProps, lastPostAtPerChannel]: FilterAndSortMyChannelsArgs): MyChannelModel[] => {
     const mentions: MyChannelModel[] = [];
     const unreads: MyChannelModel[] = [];
     const mutedMentions: MyChannelModel[] = [];
+    const latestPerChannelMentions: Record<string, number> = {};
+    const latestPerChannelUnread: Record<string, number> = {};
+    const latestPerChannelMuted: Record<string, number> = {};
 
     const isMuted = (id: string) => {
         return notifyProps[id]?.mark_unread === 'mention';
     };
 
     for (const myChannel of myChannels) {
+        const id = myChannel.id;
+
         // is it a mention?
-        if (!isMuted(myChannel.id) && myChannel.mentionsCount > 0) {
+        if (!isMuted(id) && myChannel.mentionsCount > 0) {
             mentions.push(myChannel);
+            if (lastPostAtPerChannel?.[id]) {
+                latestPerChannelMentions[id] = lastPostAtPerChannel[id];
+            }
             continue;
         }
 
         // is it unread?
         if (!isMuted(myChannel.id) && myChannel.isUnread) {
             unreads.push(myChannel);
+            if (lastPostAtPerChannel?.[id]) {
+                latestPerChannelUnread[id] = lastPostAtPerChannel[id];
+            }
             continue;
         }
 
         // is it a muted mention?
         if (isMuted(myChannel.id) && myChannel.mentionsCount > 0) {
             mutedMentions.push(myChannel);
+            if (lastPostAtPerChannel?.[id]) {
+                latestPerChannelMuted[id] = lastPostAtPerChannel[id];
+            }
             continue;
         }
     }
 
     // Sort
-    mentions.sort(mostRecentFirst);
-    unreads.sort(mostRecentFirst);
-    mutedMentions.sort(mostRecentFirst);
+    mentions.sort(mostRecentFirst.bind(null, lastPostAtPerChannel ? latestPerChannelMentions : undefined));
+    unreads.sort(mostRecentFirst.bind(null, lastPostAtPerChannel ? latestPerChannelUnread : undefined));
+    mutedMentions.sort(mostRecentFirst.bind(null, lastPostAtPerChannel ? latestPerChannelMuted : undefined));
```

Comment:
Wouldn't it be simpler to use for all the `lastPostAtPerChannel` and in the `mostRecentFirst` function check whether both exist in the array? I don't think the performance loss is so much, but I feel the code complexity is worth it.